pkg/cloud/ibmcloud/deployer.go

```diff
@@ -89,3 +93,37 @@ func (d *DeploymentClient) GetKubeConfig(cluster *clusterv1.Cluster, master *clu
 	}
 	return strings.TrimSpace(parts[1]), nil
 }
+
+func getSSHKeyFile(homeDir string) string {
+	defaultKey := homeDir + "/.ssh/id_ibmcloud"
+	envPrivateKey, ok := os.LookupEnv("IBMCLOUD_HOST_SSH_PRIVATE_FILE")
+	if !ok || (0 == strings.Compare(envPrivateKey, "")) {
+		return defaultKey
+	}
+	klog.V(3).Infof("Found environment variable SSH private file %s ", envPrivateKey)
+	exist, _ := FileExists(envPrivateKey)
+	if exist {
+		return envPrivateKey
+	}
+	privateKey := filepath.Join(GetCurrentPath(), "examples", "ibmcloud", envPrivateKey)
```

Comment:
Did you mean `envPrivateKey` is getting from relative path related "examples/ibmcloud"?

Reply:
Yes, user run `generate-yaml.sh` under examples/ibmcloud by doc. If it is a relative path the key files should be generated under relative path to `examples/ibmcloud`.

Reply:
It seems not easy to explain this behavior. The `clusterctl` tries to find a file relative to `examples/ibmcloud`. This seems weird to me. Do you have any other ideas for this?

Reply:
Let's take a example to discuss here, it may help us. 
Code is under /gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/
1.  User export a relative path.
```
export IBMCLOUD_HOST_SSH_PRIVATE_FILE=mykey
```
2. User run `generate-yaml.sh`
```
cd examples/ibmcloud
./generate-yaml.sh [options] <path/to/clouds.yaml> <provider os: [centos,ubuntu,coreos]>
cd ../..
```
3. The files `mykey` and `mykey.pub` were created under path.
```
root@gummite1:/gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud# ll /gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud/mykey*
-rw------- 1 root root 1679 Jun 15 20:51 /gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud/mykey
-rw-r--r-- 1 root root  408 Jun 15 20:51 /gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud/mykey.pub
root@gummite1:/gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud#
```
4. When you run `clusterctl` you are under `/gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl`
To `clusterctl` the files `mykey` and `mykey.pub` should be under the relative path `examples/ibmcloud` .

5. The code is get the current path for `clusterctl`, then append the relative path `examples/ibmcloud`, then append the relative path user defined.

`/gopath/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl` + `examples/ibmcloud` + $envPrivateKey

6. If it is absolute path, it has been checked.

Reply:
Consider this is a common way to handle customize key file, and user may run those commands under any directory, use current path plus relative path user defined is right way.  
So remove hard-code path  `examples/ibmcloud` to make it more flexible.
Thanks for @xunpan comments and lls review.